### PR TITLE
Fix PyTorch training loop

### PR DESCRIPTION
**Pull request by gorold**
*Description of changes:*
Minor bug in the pytorch training loop.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Review by jaheba — APPROVED** [review]
`ruff` removed the check that failed in the newer version, so this should be good to go.

File: src/gluonts/torch/model/estimator.py [hunk 1]
```diff
@@ -174,8 +174,8 @@ def train_model(
 
         validation_data_loader = None
 
-        with env._let(max_idle_transforms=max(len(training_data), 100)):
-            if validation_data is not None:
+        if validation_data is not None:
+            with env._let(max_idle_transforms=max(len(validation_data), 100)):
```
Comment (jaheba) [inline src/gluonts/torch/model/estimator.py:178]: Something I'm wondering, shouldn't there never be idle transforms, since the instance splitter always returns one window for validation per ts?
